src/oge/emissions.py

```diff
@@ -378,32 +378,47 @@ def adjust_emissions_for_biomass(df):
     return df
 
 
-def calculate_co2e_mass(df, year, gwp_horizon=100, ar5_climate_carbon_feedback=True):
+def calculate_co2e_mass(df, version, gwp_horizon=100, ar5_climate_carbon_feedback=True):
     """Calculate CO2-equivalent emissions from CO2, CH4, and N2O. This is done by
     choosing one of the IPCC's emission factors for CH4 and N2O.
 
+    Note that `version` parameter controls the version of the IPCC version that will be
+    chosen. If `version` is made of numerals only then `version` is interpreted as a
+    year the most recent AR that was available in the given year is picked. If
+    `version` is mage of letters only or is alphanumeric then `version` will be
+    interpreted as the AR.
+
     If the `fuel_consumed_for_electricity_units` column is available, we also compute
     the adjusted emissions.
     """
     df_gwp = load_data.load_ipcc_gwp()
 
-    # use the most recent AR that was available in the given year
-    ipcc_version = df_gwp.loc[df_gwp["year"] <= year, "year"].max()
+    if isinstance(version, int) and version in df_gwp["year"].unique():
+        year = version
+        # use the most recent AR that was available in the given year
+        ipcc_version = df_gwp.loc[df_gwp["year"] <= year, "year"].max()
 
-    # use the most recent AR that was available in the given year
-    most_recent_AR_year = df_gwp.loc[df_gwp["year"] <= year, "year"].max()
+        # use the most recent AR that was available in the given year
+        most_recent_AR_year = df_gwp.loc[df_gwp["year"] <= year, "year"].max()
 
-    # identify the AR
-    ipcc_version = list(
-        df_gwp.loc[df_gwp["year"] == most_recent_AR_year, "ipcc_version"].unique()
-    )
-    if len(ipcc_version) > 1:
-        if "AR5" in ipcc_version and ar5_climate_carbon_feedback:
-            ipcc_version = "AR5_cc"
-        elif "AR5" in ipcc_version and not ar5_climate_carbon_feedback:
-            ipcc_version = "AR5_cc"
+        # identify the AR
+        ipcc_version = list(
+            df_gwp.loc[df_gwp["year"] == most_recent_AR_year, "ipcc_version"].unique()
+        )
+        if len(ipcc_version) > 1:
+            if "AR5" in ipcc_version and ar5_climate_carbon_feedback:
+                ipcc_version = "AR5_cc"
+            elif "AR5" in ipcc_version and not ar5_climate_carbon_feedback:
+                ipcc_version = "AR5_cc"
+        else:
+            ipcc_version = ipcc_version[0]
+    elif isinstance(version, str) and version in df_gwp["ipcc_version"].unique():
```

Comment:
Here, if someone just enters "AR5" and sets ar5_climate_carbon_feedback=True, this will not pass the second test.

Reply:
The `ar5_climate_carbon_feedback` is not used when `version` is a string. If you pass `AR5` then `ipcc_version` is set to `AR5`, if you pass `AR5_cc` then `ipcc_version` is set to `AR5_cc`

Reply:
Got it. Can we add this explanation to the docstring so that someone is not expecting to get the AR5_cc value back if they set this to True?

Maybe in the docstring we should also enumerate all valid values of version, eg an int >=1995 or a string in SAR, TAR, AR4, AR5, AR5_cc, or AR6.

Reply:
I have updated the docstring. Let me know what you think.